dts/arm/yaml/st,stm32-otgfs.yaml

```diff
@@ -0,0 +1,56 @@
+---
+title: STM32 OTGFS
+id: st,stm32-otgfs
+version: 0.1
+
+description: >
+    This binding gives a base representation of the STM32 OTGFS controller
+
+inherits:
+  - !include usb.yaml
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "st,stm32-otgfs"
+
+  - reg:
+      type: array
+      description: mmio register space
+      generation: define
+      category: required
+
+  - interrupts:
+      type: array
+      category: required
+      description: required interrupts
+      generation: define
+
+  - interrupt-names:
+      type: stringlist
+      category: optional
+      description: readable string describing the interrupts
+      generation: define
+
+  - clocks:
+      type: array
+      category: required
+      description: Clock gate control information
+      generation: define
+
+  - endpoints-number:
```

Comment:
I guess `endpoints-number` can be moved to dts/common/yaml/usb.yaml

Reply:
done

Reply:
do we need to duplicate it here if its now in common yaml?

Reply:
I missed it, stm32 usb driver needs the number of bidirectional endpoints to configure the FIFO (?), so is platform  specific. s/endpoints-number/num-bidirectional-endpoints ?

Reply:
The driver needs the number of IN endpoints to configure the FIFO (there is a TX-IN FIFO for each IN endpoint). The driver also use this number to enable IN and OUT endpoints. For example, stm32f4 otgfs controller has 4 OUT and 4 IN endpoints (including EP0).

Reply:
So do we always assume a bidirectional control endpoint, and the having properties for number of bi-directional, number of IN, and number of OUT endpoints?

Reply:
All USB driver implementations I have seen, which are using STM32 HAL (HAL examples, embed-os),  use the number of bidirectional endpoints to initialize the controller. Then they enable and configure the endpoints they actually use. At least for the STM32 USB one property for bidirectional endpoints is enough for now. I don't know if it is enough for USB controllers from other vendors.
Truth be told, I don't have any prior experience with USB, I just started studying it, so I consider my involvement with STM32 USB driver my baby steps.

Reply:
I think we should writeup the DT from the HW point of view as best as possible and not as much form the driver point of view.  While we take both into account, the DT should describe the HW, so any different driver implementation should be able to utilize it.  Thus specifying the endpoints they way I mentioned.

It looks like we have Intel Quark, STM32, and Silabs that utilize the DWC USB IP.  So seeing what they all need/do is the best way foward.  The SiLabs EFM32WG seems to support EP0 + 6IN + 6 OUT.  Intel Quark C1000 says 1 control (i assume this is ep0), 3 bi, 2 IN.  The STM32 F4x5/F4x7/F4x9 Says 1 bidirection (ep0), 3 IN, 3 OUT.

Reply:
I will add the properties for number of bi-directional, number of IN, and number of OUT endpoints.